docs/testing-guide/state-management.md

```diff
@@ -0,0 +1,100 @@
+# State Management
+
+`algorand-python-testing` provides tools to test state-related abstractions in Algorand smart contracts. This guide covers global state, local state, boxes, and scratch space management.
+
+```{testsetup}
+import algopy
+import algopy_testing
+
+# Create the context manager for snippets below
+ctx_manager = algopy_testing_context()
+
+# Enter the context
+context = ctx_manager.__enter__()
```

Comment:
avoid examples with `__enter__` being called explicitly, either have a pytest fixture to provide the context, or just have a `with algopy_testing_context()`